tests/test_data_utils/test_select_values.py

```diff
@@ -59,6 +59,20 @@ def test_unsupported_data_type(self):
         with self.assertRaises(TypeError):
             select_values(["not", "a", "valid", "input"], 'column1', ['A'])
 
+    def test_select_values_dataframe_nonexistent_values(self):
```

Comment:
@fangliu117 , please 
1- add a test for correct behavior when you pass an andata object.
2- In your assertion, make sure the all returned "values" are within the list of values you provided, e.g., if you only select the value 'x' from an observation, all returned cells will have have the value 'X' only. 
3- Please edit the tests that works on the dataframe to check that the returned values are correct as well. Currently I see that check is the number of returned cells is correct.